__Exploratory Data Analysis__ 

This project is a continuation of the layoffs data I cleaned in the previous project

__Highlights from the Project__

- I utilized aggregation functions such as group by, order by, limit etc
- I utilized advanced sql functions such as common table expressions (CTEs) including using two window functions within a single query, and window functions.

__Highlights on EDA__

- The data covers layoffs from companies aross the world from March 2020 through March 2023
- The post-covid period saw the highest layoffs with 2022 and 2023 showing exponential increase in total layoffs as evidenced by the running total in this project
- The consumer industry spanning companies such as retailers and tech startups(Uber, Tiktok etc) and bigTech(FAANG) had the highest layoffs, while manufacturing exhibited the least layoffs.
- While the bigTech companies in the US had the highest layoffs, they only laid off a tiny percentage of their employees (2-12%), whereas there were other companies that laid off 100% of their employees.
- Companies with unknown funding had the highest layoffs, closely followed by big companies in post-IPO growth stage. Additionally, companies in Series C, followed by B, then A had a significnatly high amount of layoffs, whereas those in the Series I,G,H,J had the least layoffs. Probably indicating that such companies are less likely to layoff many employees.
- Most companies in this data were from USA, India, Netherlands, and Canada.

__Next Steps__

- I will further analyze this data in python and check for any significant association between the industry, company, funding amount with the total layoffs.
- I will build a machine Learning model that classifies companies as "Low,Medium,High" risk of laying off employees.
- I will create a dashboard in PowerBI to highlight the key insights from the analysis




In [1]:
!pip install ipython-sql pymysql;

For security purposes, I deleted one cell here that established a connection to mysql workbech in my local pc.

For replicability, please download the data (layoffs_staging2) I have used here as attached in this repository, and create a database in your mysql called 'world_layoffs' then add this data via the data import wizard.

To set up a connection to your mysql, please use the following code in a jupyter notebook cell 

%sql mysql+pymysql://myuser:mypassword@localhost/mydatabase

Replace the credentials such as user, password, localhost, and database accordingly.


In [5]:
%%sql
SELECT *
FROM layoffs_staging2
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.


[(' Included Health', 'SF Bay Area', 'Healthcare', None, '0.06', datetime.date(2022, 7, 25), 'Series E', 'United States', 272),
 ('&Open', 'Dublin', 'Marketing', 9, '0.09', datetime.date(2022, 11, 17), 'Series A', 'Ireland', 35),
 ('#Paid', 'Toronto', 'Marketing', 19, '0.17', datetime.date(2023, 1, 27), 'Series B', 'Canada', 21),
 ('100 Thieves', 'Los Angeles', 'Consumer', 12, None, datetime.date(2022, 7, 13), 'Series C', 'United States', 120),
 ('10X Genomics', 'SF Bay Area', 'Healthcare', 100, '0.08', datetime.date(2022, 8, 4), 'Post-IPO', 'United States', 242),
 ('1stdibs', 'New York City', 'Retail', 70, '0.17', datetime.date(2020, 4, 2), 'Series D', 'United States', 253),
 ('2TM', 'Sao Paulo', 'Crypto', 90, '0.12', datetime.date(2022, 6, 1), 'Unknown', 'Brazil', 250),
 ('2TM', 'Sao Paulo', 'Crypto', 100, '0.15', datetime.date(2022, 9, 1), 'Unknown', 'Brazil', 250),
 ('2U', 'Washington D.C.', 'Education', None, '0.2', datetime.date(2022, 7, 28), 'Post-IPO', 'United States', 426),
 ('54gene', 'Washington D.C.', 'Healthcare', 95, '0.3', datetime.date(2022, 8, 29), 'Series B', 'United States', 44)]

To provide context into the period under review, I will pull the max and min date

In [7]:
%%sql
SELECT min(`date`) as min_date, max(`date`) as max_date
FROM layoffs_staging2;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
1 rows affected.


[(datetime.date(2020, 3, 11), datetime.date(2023, 3, 6))]

In [9]:
result = %sql SELECT min(`date`) as min_date, max(`date`) as max_date FROM layoffs_staging2;
result
#The data runs from 11 march, 2020 through 5 June, 2024.

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
1 rows affected.


[(datetime.date(2020, 3, 11), datetime.date(2023, 3, 6))]

I will first explore the total laid off column to understand its general outlook

First, explore the descriptives such as max, min, mean etc employees laid off

In [11]:
%%sql result_df <<
SELECT MAX(total_laid_off) AS max_layoffs,
       MIN(total_laid_off) AS min_layoffs,
       AVG(total_laid_off) AS avg_layoffs,
       SUM(total_laid_off) AS total_overall_layoffs
FROM layoffs_staging2;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
1 rows affected.
Returning data to local variable result_df


In [13]:
result_df
#For the period under review, the companies laid off at least 237 employees per layoff occasion, 
# with some company laying off a whooping 12000 employees, and some company laying off 3 employees.

[(12000, 3, Decimal('237.2659'), Decimal('383659'))]

It will be interesting to understand if there are companies that laid off all of their employees

In [15]:
%%sql laid_off_df <<
SELECT company, total_laid_off
FROM layoffs_staging2
WHERE percentage_laid_off = 1
ORDER BY total_laid_off DESC
limit 10;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable laid_off_df


In [17]:
laid_off_df

[('Katerra', 2434),
 ('Butler Hospitality', 1000),
 ('Deliv', 669),
 ('Jump', 500),
 ('SEND', 300),
 ('Stoqo', 250),
 ('HOOQ', 250),
 ('Stay Alfred', 221),
 ('Britishvolt', 206),
 ('Planetly', 200)]

For the companies that laid off 100% of its employees (so to say the company went under!), the top number
laid off were 2434 employees by Keterra, followed by Butler hospitality.

It will be interesting to know which companies had the highest layoffs

In [19]:
%%sql company_layoffs <<
SELECT company, country, SUM(total_laid_off) AS total_laid_off
FROM layoffs_staging2
GROUP BY company, country
ORDER BY total_laid_off DESC
LIMIT 10;


 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable company_layoffs


In [21]:
company_layoffs

[('Amazon', 'United States', Decimal('18150')),
 ('Google', 'United States', Decimal('12000')),
 ('Meta', 'United States', Decimal('11000')),
 ('Salesforce', 'United States', Decimal('10090')),
 ('Philips', 'Netherlands', Decimal('10000')),
 ('Microsoft', 'United States', Decimal('10000')),
 ('Ericsson', 'Sweden', Decimal('8500')),
 ('Uber', 'United States', Decimal('6700')),
 ('Dell', 'United States', Decimal('6650')),
 ('Booking.com', 'Netherlands', Decimal('4375'))]

Amazon (18,150), Google(12000), Meta(11000), Salesforce(10,090), and Microsoft(10000) top the list of most layoffs within the period reviewed

It looks like tech companies laid off the most employees, but having a tally of layoffs per industry would provide more context

In [23]:
%%sql industry_layoffs <<
SELECT industry, SUM(total_laid_off) AS total_laid_off
FROM layoffs_staging2
GROUP BY industry
ORDER BY total_laid_off DESC
LIMIT 10;


 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable industry_layoffs


In [25]:
industry_layoffs

[('Consumer', Decimal('45182')),
 ('Retail', Decimal('43613')),
 ('Other', Decimal('36289')),
 ('Transportation', Decimal('33748')),
 ('Finance', Decimal('28344')),
 ('Healthcare', Decimal('25953')),
 ('Food', Decimal('22855')),
 ('Real Estate', Decimal('17565')),
 ('Travel', Decimal('17159')),
 ('Hardware', Decimal('13828'))]

Consumer and Retail industry had the highest layoffs 45182, and 43613 in total

I would want to know which industry classification for the companies identified earlier as the the leading with layoffs 


In [27]:
%%sql company_industry <<
SELECT company, industry
FROM layoffs_staging2
WHERE company IN ('Google', 'Amazon', 'Microsoft', 'Meta', 'Salesforce', 'Microsoft');


 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable company_industry


In [29]:
company_industry

[('Amazon', 'Retail'),
 ('Amazon', 'Retail'),
 ('Amazon', 'Retail'),
 ('Google', 'Consumer'),
 ('Meta', 'Consumer'),
 ('Microsoft', 'Other'),
 ('Salesforce', 'Sales'),
 ('Salesforce', 'Sales'),
 ('Salesforce', 'Sales'),
 ('Salesforce', 'Sales')]

Google and meta are classified as consumer, amazon as retail, and micrososft as other. 

It would be interesting to know the schema behind the classification of the industry. For instance, why was there no classification for technology

Manufacturing had the least layoffs

Next, explore the distribution of total layoffs per country

In [31]:
%%sql country_layoffs <<
SELECT country, SUM(total_laid_off) total_layoffs ,count(country) as no_of_companies
FROM layoffs_staging2
GROUP BY country
ORDER BY total_layoffs desc
LIMIT 10; 

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable country_layoffs


In [33]:
country_layoffs

[('United States', Decimal('256559'), 1294),
 ('India', Decimal('35993'), 140),
 ('Netherlands', Decimal('17220'), 12),
 ('Sweden', Decimal('11264'), 17),
 ('Brazil', Decimal('10391'), 69),
 ('Germany', Decimal('8701'), 62),
 ('United Kingdom', Decimal('6398'), 58),
 ('Canada', Decimal('6319'), 90),
 ('Singapore', Decimal('5995'), 26),
 ('China', Decimal('5905'), 13)]

As expected the united states had the highest layoffs 256,559 (about 67% of all layoffs in the period reviewed) , followed by India(35,993) and Netherlands (17,220)

Most companie were from the US and India

Aside from the total layoffs, it would be interesting to understand, how the industry and companies tallied in the percentage of employees laid off

First, explore the industry

In [35]:
%%sql industry_percentage_layoffs <<
SELECT industry, sum(percentage_laid_off)
FROM layoffs_staging2
GROUP BY industry
ORDER BY 2 desc
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable industry_percentage_layoffs


In [ ]:
industry_percentage_layoffs

Interestingly, while the consuemr sector topped in the cummulative total layoffs, 

It seems that it did not layoff the largest percentage of its staff.

The finance and healthcare sectors had the highest percentage of employees laid off.

As consisntent with the cummulative layoffs, the manufacturing sector had the least percentage of its stuff laid off

How about the tally per company?


In [37]:
%%sql company_percentage_layoffs << 
SELECT company, sum(percentage_laid_off)
FROM layoffs_staging2
GROUP BY company
ORDER BY 2 desc
LIMIT 10;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
10 rows affected.
Returning data to local variable company_percentage_layoffs


In [39]:
company_percentage_layoffs 

[('Service', 2.0),
 ('Airy Rooms', 1.7),
 ('Medly', 1.66),
 ('Pollen', 1.6400000000000001),
 ('Rubius', 1.5699999999999998),
 ('The Wing', 1.5),
 ('Propzy', 1.5),
 ('Airlift', 1.31),
 ('Zeus Living', 1.26),
 ('WanderJaunt', 1.23)]

Interestingly the top companies with the most employee layoffs do not top the list of the companies with the highest percentage of employees laid off

Then it will be interesting to know which percentage of employees were laid off for the companies with the most layoffs

In [41]:
%%sql top_company_layoffs_percentage <<

with company_percentage_laid_off_cte as
(select company, sum(percentage_laid_off)
from layoffs_staging2
group by company
order by 2 desc)
 select *
 from company_percentage_laid_off_cte
 where company in ('Google','Amazon','Microsoft','Meta','Salesforce','Microsoft');

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
5 rows affected.
Returning data to local variable top_company_layoffs_percentage


In [43]:
top_company_layoffs_percentage

[('Meta', 0.13),
 ('Salesforce', 0.1285),
 ('Google', 0.06),
 ('Amazon', 0.05),
 ('Microsoft', 0.05)]

Interesting, the companies laid off between 5% and 13% of its stuff, with Meta and Salesforce topping that list

Next, I will explore the total layoffs tally per month and per year

In [45]:
%%sql layoffs_per_year <<
SELECT YEAR(`date`), SUM(total_laid_off)
FROM layoffs_staging2 
GROUP BY YEAR(`date`)
ORDER BY 2 DESC;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
5 rows affected.
Returning data to local variable layoffs_per_year


In [47]:
layoffs_per_year

[(2022, Decimal('160661')),
 (2023, Decimal('125677')),
 (2020, Decimal('80998')),
 (2021, Decimal('15823')),
 (None, Decimal('500'))]

You would expect 2020, due to the covid scenario to have the most layoffs, but interestingly,
2022 instead, had the highest layoffs, totalling to 160,661, closely followed by 2023 with 125,677 layoffs. 
Even more interestingly, the data has just 3 months of 2023, yet the total layoffs are just short of the total layoffs in 2022. 

Next, it would be interresting to know if there a specific month of the eyar where most layoffs are most likely to occur?

In [49]:
%%sql monthly_layoffs << 
select monthname(`date`), sum(total_laid_off)
from layoffs_staging2
group by 1
order by 2 desc; 

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
13 rows affected.
Returning data to local variable monthly_layoffs


In [51]:
monthly_layoffs

[('January', Decimal('92037')),
 ('November', Decimal('55758')),
 ('February', Decimal('41046')),
 ('May', Decimal('38689')),
 ('April', Decimal('31099')),
 ('June', Decimal('27455')),
 ('July', Decimal('23415')),
 ('March', Decimal('19859')),
 ('October', Decimal('17878')),
 ('August', Decimal('16891')),
 ('December', Decimal('12381')),
 ('September', Decimal('6651')),
 (None, Decimal('500'))]

Looks like most layoffs take place beginning of Q1 (January) and end of Q3 (November)

In [ ]:
#Finally, it will be interesting to know how the total layoffs tally with the stage of funding of the company

In [53]:
%%sql stage_layoffs <<
select stage, sum(total_laid_off) as total_layoffs
from layoffs_staging2
group by stage
order by 2 desc;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
17 rows affected.
Returning data to local variable stage_layoffs


In [55]:
stage_layoffs

[('Post-IPO', Decimal('204132')),
 ('Unknown', Decimal('40716')),
 ('Acquired', Decimal('27576')),
 ('Series C', Decimal('20017')),
 ('Series D', Decimal('19225')),
 ('Series B', Decimal('15311')),
 ('Series E', Decimal('12697')),
 ('Series F', Decimal('9932')),
 ('Private Equity', Decimal('7957')),
 ('Series H', Decimal('7244')),
 ('Series A', Decimal('5678')),
 ('Series G', Decimal('3697')),
 ('Series J', Decimal('3570')),
 ('Series I', Decimal('2855')),
 ('Seed', Decimal('1636')),
 ('Subsidiary', Decimal('1094')),
 (None, Decimal('322'))]

Interestingly most layoffs came from companies that are already listed in stock markets, closely followed companies whose funding stage is unknown, and then those that are acquired

Interestingly those in seed stage have very low layoffs, probably because the company is still at the stage of proof of concept

Does the same trend hold, for the percentage to be laid off?

In [57]:
%%sql stage_percentage_layoffs <<
select stage, sum(percentage_laid_off) as total_percentage_laid_off
 from layoffs_staging2
 group by stage
 order by 2 desc;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
17 rows affected.
Returning data to local variable stage_percentage_layoffs


In [59]:
stage_percentage_layoffs

[('Unknown', 76.53000000000002),
 ('Series B', 68.61999999999995),
 ('Post-IPO', 48.718500000000034),
 ('Series C', 46.36999999999998),
 ('Series A', 42.61999999999997),
 ('Seed', 32.980000000000004),
 ('Series D', 30.664999999999996),
 ('Acquired', 26.743999999999993),
 ('Series E', 15.719999999999994),
 ('Series F', 5.4399999999999995),
 ('Private Equity', 4.61),
 ('Series G', 2.0900000000000003),
 ('Series H', 2.04),
 (None, 1.16),
 ('Series I', 0.5800000000000001),
 ('Series J', 0.57),
 ('Subsidiary', 0.16999999999999998)]

The unclassified stage of funding companies laid off the highest percentage of their employees,

Closely followed by those in Series B, post-IPO,Series C, then Series A. There is a trend that the companies in Series A, B,then C are likley to layoff employees

Looks like most companies start to struggle as from Series B stage of funding.

Conversely, companie sin advanced funding stages, such as series H,I,J laid off the least percentage of their employees

Next, I will now create a summary highligting the running total of cummulative layoffs per month per company.

So to say, create a rolling sum of layoffs across the period under review, starting with per month

In [68]:
%%sql monthly_total_layoffs <<
SELECT SUBSTRING(`date`,1,7) AS `MONTH`, SUM(total_laid_off) as monthly_total
FROM layoffs_staging2
WHERE SUBSTRING(`date`,1,7) IS NOT NULL
GROUP BY `MONTH`
ORDER BY 1 ASC;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
36 rows affected.
Returning data to local variable running_total_layoffs


In [70]:
monthly_total_layoffs

[('2020-03', Decimal('9628')),
 ('2020-04', Decimal('26710')),
 ('2020-05', Decimal('25804')),
 ('2020-06', Decimal('7627')),
 ('2020-07', Decimal('7112')),
 ('2020-08', Decimal('1969')),
 ('2020-09', Decimal('609')),
 ('2020-10', Decimal('450')),
 ('2020-11', Decimal('237')),
 ('2020-12', Decimal('852')),
 ('2021-01', Decimal('6813')),
 ('2021-02', Decimal('868')),
 ('2021-03', Decimal('47')),
 ('2021-04', Decimal('261')),
 ('2021-06', Decimal('2434')),
 ('2021-07', Decimal('80')),
 ('2021-08', Decimal('1867')),
 ('2021-09', Decimal('161')),
 ('2021-10', Decimal('22')),
 ('2021-11', Decimal('2070')),
 ('2021-12', Decimal('1200')),
 ('2022-01', Decimal('510')),
 ('2022-02', Decimal('3685')),
 ('2022-03', Decimal('5714')),
 ('2022-04', Decimal('4128')),
 ('2022-05', Decimal('12885')),
 ('2022-06', Decimal('17394')),
 ('2022-07', Decimal('16223')),
 ('2022-08', Decimal('13055')),
 ('2022-09', Decimal('5881')),
 ('2022-10', Decimal('17406')),
 ('2022-11', Decimal('53451')),
 ('2022-12', Decimal('10329')),
 ('2023-01', Decimal('84714')),
 ('2023-02', Decimal('36493')),
 ('2023-03', Decimal('4470'))]

Next, I would like to create a rolling total of the total layoffs for the period under study, so that the gradual increase or decreaase in the total come sout more clearly
I will utilize a cte,and windows function

In [90]:
%%sql rolling_total << 
WITH rolling_total_cte AS
(SELECT SUBSTRING(`date`,1,7) AS `YEAR-MONTH`, SUM(total_laid_off) as monthly_total
FROM layoffs_staging2
WHERE SUBSTRING(`date`,1,7) IS NOT NULL
GROUP BY `YEAR-MONTH`
ORDER BY 1 ASC
)
SELECT `YEAR-MONTH`, monthly_total,SUM(monthly_total) OVER(ORDER BY `YEAR-MONTH`) as running_total
FROM rolling_total_cte;


 * mysql+pymysql://root:***@localhost:3308/world_layoffs
36 rows affected.
Returning data to local variable rolling_total


In [92]:
rolling_total

[('2020-03', Decimal('9628'), Decimal('9628')),
 ('2020-04', Decimal('26710'), Decimal('36338')),
 ('2020-05', Decimal('25804'), Decimal('62142')),
 ('2020-06', Decimal('7627'), Decimal('69769')),
 ('2020-07', Decimal('7112'), Decimal('76881')),
 ('2020-08', Decimal('1969'), Decimal('78850')),
 ('2020-09', Decimal('609'), Decimal('79459')),
 ('2020-10', Decimal('450'), Decimal('79909')),
 ('2020-11', Decimal('237'), Decimal('80146')),
 ('2020-12', Decimal('852'), Decimal('80998')),
 ('2021-01', Decimal('6813'), Decimal('87811')),
 ('2021-02', Decimal('868'), Decimal('88679')),
 ('2021-03', Decimal('47'), Decimal('88726')),
 ('2021-04', Decimal('261'), Decimal('88987')),
 ('2021-06', Decimal('2434'), Decimal('91421')),
 ('2021-07', Decimal('80'), Decimal('91501')),
 ('2021-08', Decimal('1867'), Decimal('93368')),
 ('2021-09', Decimal('161'), Decimal('93529')),
 ('2021-10', Decimal('22'), Decimal('93551')),
 ('2021-11', Decimal('2070'), Decimal('95621')),
 ('2021-12', Decimal('1200'), Decimal('96821')),
 ('2022-01', Decimal('510'), Decimal('97331')),
 ('2022-02', Decimal('3685'), Decimal('101016')),
 ('2022-03', Decimal('5714'), Decimal('106730')),
 ('2022-04', Decimal('4128'), Decimal('110858')),
 ('2022-05', Decimal('12885'), Decimal('123743')),
 ('2022-06', Decimal('17394'), Decimal('141137')),
 ('2022-07', Decimal('16223'), Decimal('157360')),
 ('2022-08', Decimal('13055'), Decimal('170415')),
 ('2022-09', Decimal('5881'), Decimal('176296')),
 ('2022-10', Decimal('17406'), Decimal('193702')),
 ('2022-11', Decimal('53451'), Decimal('247153')),
 ('2022-12', Decimal('10329'), Decimal('257482')),
 ('2023-01', Decimal('84714'), Decimal('342196')),
 ('2023-02', Decimal('36493'), Decimal('378689')),
 ('2023-03', Decimal('4470'), Decimal('383159'))]

The observed trend is that as from 2020 through December 2021, the layoffs were modest as compared to the period running from January 2022 through March 2023

Lastly, it would be of interest to know how each company laid off employees across the years. Also, show the rank such that I can see the worst year where the company laid off the most people at the top

In [117]:
%%sql company_year_rank <<
WITH company_yearly_rank AS (
    SELECT company, YEAR(`date`) AS years, SUM(total_laid_off) AS total_layoffs
    FROM layoffs_staging2
    GROUP BY company, YEAR(`date`)
),
company_sub_rank AS (
    SELECT *, DENSE_RANK() OVER(PARTITION BY years ORDER BY total_layoffs DESC) AS ranking
    FROM company_yearly_rank
    WHERE years IS NOT NULL
    ORDER BY ranking ASC
)
SELECT *
FROM company_sub_rank
WHERE ranking <= 5;

 * mysql+pymysql://root:***@localhost:3308/world_layoffs
22 rows affected.
Returning data to local variable company_year_rank


In [119]:
company_year_rank

[('Uber', 2020, Decimal('7525'), 1),
 ('Bytedance', 2021, Decimal('3600'), 1),
 ('Meta', 2022, Decimal('11000'), 1),
 ('Google', 2023, Decimal('12000'), 1),
 ('Booking.com', 2020, Decimal('4375'), 2),
 ('Katerra', 2021, Decimal('2434'), 2),
 ('Amazon', 2022, Decimal('10150'), 2),
 ('Microsoft', 2023, Decimal('10000'), 2),
 ('Groupon', 2020, Decimal('2800'), 3),
 ('Zillow', 2021, Decimal('2000'), 3),
 ('Cisco', 2022, Decimal('4100'), 3),
 ('Ericsson', 2023, Decimal('8500'), 3),
 ('Swiggy', 2020, Decimal('2250'), 4),
 ('Instacart', 2021, Decimal('1877'), 4),
 ('Peloton', 2022, Decimal('4084'), 4),
 ('Amazon', 2023, Decimal('8000'), 4),
 ('Salesforce', 2023, Decimal('8000'), 4),
 ('Airbnb', 2020, Decimal('1900'), 5),
 ('WhiteHat Jr', 2021, Decimal('1800'), 5),
 ('Carvana', 2022, Decimal('4000'), 5),
 ('Philips', 2022, Decimal('4000'), 5),
 ('Dell', 2023, Decimal('6650'), 5)]

The above table show which company, for each year, had the top 5 most lay_offs 

--END